new-components/enrichers/custom-annotation/internal/annotation/annotation.go

```diff
@@ -0,0 +1,108 @@
+package annotation
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+	"slices"
+
+	"github.com/smithy-security/pkg/env"
+
+	"github.com/smithy-security/smithy/sdk/component"
+	ocsf "github.com/smithy-security/smithy/sdk/gen/com/github/ocsf/ocsf_schema/v1"
+)
+
+type (
+	// Conf contains the annotation configuration.
+	Conf struct {
+		AnnotationName   string
+		AnnotationValues map[string]any
+	}
+
+	customAnnotator struct {
+		conf *Conf
+	}
+)
+
+// NewConf initialises the enricher configuration from environment variables.
+func NewConf() (*Conf, error) {
+	var (
+		cfg = &Conf{
+			AnnotationValues: make(map[string]any),
+		}
+		err error
+	)
+
+	cfg.AnnotationName, err = env.GetOrDefault(
+		"CUSTOM_ANNOTATION_NAME",
+		"",
+		env.WithDefaultOnError(true),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	annotationValues, err := env.GetOrDefault("CUSTOM_ANNOTATION_VALUES", "{}")
+	if err != nil {
+		return nil, err
+	}
+
+	if len(annotationValues) > 0 && annotationValues != "{}" {
```

Comment:
I think we can simplify this a bit more and just unmarshall if the annotationValues is a non-empty string